frontend/src/utils/segment.js

```diff
@@ -0,0 +1,13 @@
+export default class Segment {
```

Comment:
Not sure if this will be too much work but I think it might be better to just declare this file in `.ts` so that we don't need to rewrite this file again in future. @gerhean @dcshzj any thoughts?

Reply:
I have written the classes in TypeScript now. Please take a look!